dist/bundle.js

```diff
@@ -0,0 +1,743 @@
+      target.updateColor('#e5e5ff', target.id);
+    }
+    else if (isClose(targetX, targetY, 450)) {
+      target.updateColor('#ccccff', target.id);
+    }
+    else if (isClose(targetX, targetY, 400)) {
+      target.updateColor('#b2b2ff', target.id);
+    }
+    else if (isClose(targetX, targetY, 350)) {
+      target.updateColor('#9999ff', target.id);
+    }
+    else if (isClose(targetX, targetY, 300)) {
+      target.updateColor('#7f7fff', target.id);
+    }
+    else if (isClose(targetX, targetY, 250)) {
+      target.updateColor('#6666ff', target.id);
+    }
+    else if (isClose(targetX, targetY, 200)) {
+      target.updateColor('#4c4cff', target.id);
+    }
+    else if (isClose(targetX, targetY, 150)) {
+      target.updateColor('#3232ff', target.id);
+    }
+    else if (isClose(targetX, targetY, 100)) {
+      target.updateColor('#1919ff', target.id);
+    }
+    else if (isClose(targetX, targetY, 50)) {
+      target.updateColor('#0000ff', target.id);
+    }
+  }
+
+/**
+* @PARAM target object as described in target.js
+* @RETURNS nothing
+*/
+Canvas.prototype.drawTarget = function(target) {
+  // grab svg container
+  d3.select('#'+this.type+'-canvas').append("circle")
+    .attr({
+      cx: target.x,
+      cy: target.y,
+      r: target.r,
+      color: target.color
+    });
+};
+
+Canvas.prototype.drawTargets = function(targets) {
+  // just a for loop with drawTarget
+  for( var i = 0; i < targets.length; i++ ) {
+    drawTarget(targets[i]);
+  }
+};
+
+Canvas.prototype.checkProximity = function(vector, target) {
+  return this.isClose(vector.head.x, vector.head.y, target.x, target.y, target.r);
+}
+
+function screenToMath(x,y) {
+  return [(x - 250) * 10 / 250, - (y - 250) * 10 / 250];
+}
+
+function mathToScreen(x,y) {
+  return [x * 250 / 10 + 250, - y * 250 / 10 + 250];
+}
+
+function applyMatrix(sX,sY,matrix) {
+  var matrix = matrix || [[1,3],[2,0]];
+  var math_coord = screenToMath(sX,sY),
+      applied_coord = [matrix[0][0] * math_coord[0] + matrix[0][1] * math_coord[1], matrix[1][0] * math_coord[0] + matrix[1][1] * math_coord[1]];
+  return mathToScreen(applied_coord[0],applied_coord[1]);
+}
+
+function getRandom(min,max) {
+  return Math.random() * (max - min) + min;
+}
+
+function generateTarget(matrix) {
+  var legal = false,
+      par = matrix[0][0] * matrix[1][1] - matrix[0][1] * matrix[1][0];
+  var newX, newY;
+
+  while (!legal) {
+    newX = getRandom(0,500);
+    newY = getRandom(0,500);
+    var pre = screenToMath(newX,newY);
+    var prex = (matrix[1][1] * pre[0] - matrix[0][1] * pre[1]) / par,
+        prey = (- matrix[1][0] * pre[0] + matrix[0][0] * pre[1]) / par;
+    pre = mathToScreen(prex,prey);
+
+    if (pre[0] >= 0 && pre[0] <= 500 && pre[1] >= 0 && pre[1] <= 500) {
+      legal = true;
+      var targetSettings = {
+      	x: newX,
+      	y: newY,
+      	r: 20,
+      	color: "black",
+      	isScore: false
+      };
+      var newTarget = new Target(targetSettings);
```

Comment:
Normally this would throw an error but because we have Target in the namespace above we can use its constructor.